src/main/java/org/jenkinsci/plugins/jiraext/view/Transition.java

```diff
@@ -50,27 +50,48 @@ public Transition(String transitionName)
 
     @Override
     public void perform(List<JiraCommit> jiraCommitList,
-                        AbstractBuild build, Launcher launcher, BuildListener listener)
-    {
-        try
-        {
-            for (JiraCommit jiraCommit : JiraCommit.filterDuplicateIssues(jiraCommitList))
-            {
+                        AbstractBuild build, Launcher launcher, BuildListener listener) {
+        try {
+            for (JiraCommit jiraCommit : JiraCommit.filterDuplicateIssues(jiraCommitList)) {
                 listener.getLogger().println("Transition a ticket: " + jiraCommit.getJiraTicket());
-                listener.getLogger().println("transitionName: " + transitionName);
-                try
-                {
-                    getJiraClientSvc().changeWorkflowOfTicket(jiraCommit.getJiraTicket(), transitionName);
+
+                //Make a copy of all transitions since we will be removing the ones that are done
+                List<String> transitions = Arrays.stream(StringUtils.split(transitionName, ","))
+                        .map(t -> t.trim())
+                        .collect(Collectors.toList());
+                boolean didAnyTransition = false;
+
+                //Loop through all transitions until one works
+                while (transitions.size() > 0) {
+                    Iterator<String> transitionIter = transitions.iterator();
+                    boolean didTransition = false;
+                    while (transitionIter.hasNext()) {
+                        String transition = transitionIter.next();
+                        try {
+                            getJiraClientSvc().changeWorkflowOfTicket(jiraCommit.getJiraTicket(), transition);
+
+                            //Transition worked. Remove it from the list, and try the next one
+                            listener.getLogger().println("Performed transition: " + transition);
+                            transitionIter.remove();
+                            didAnyTransition = true;
+                            didTransition = true;
+                        } catch (Throwable t) {
+                            //Ignore and try next transition
+                        }
+                    }
+                    if (!didTransition) {
+                        //Did not perform any transitions this round. We are done!
+                        transitions.clear();
+                    }
                 }
-                catch (Throwable t)
-                {
-                    listener.getLogger().println("ERROR Updating JIRA, continuing");
-                    t.printStackTrace(listener.getLogger());
+
+                if (!didAnyTransition) {
+                    //No transitions were done. Show error message
+                    listener.getLogger().println("ERROR Updating JIRA with transitions [" + transitionName + "], continuing");
```

Comment:
This does not log information on what the type of error was -- none of the provided transitions were valid for the given ticket.

Why is 'no transitions valid' considered an error condition while 'this transition is not valid' is not an error condition?